implementing search to lsq.

sarveshms opened the pull request:
When user wants to search by title, tags, person can do that by
passing input and search. It will fetch result based on the input.

sarveshms committed:
implementing search to lsq.

When user wants to search by title, tags, person can do that by
passing input and search. It will fetch result based on the input.

shrayasr commented on app.py:
```diff
@@ -8,8 +8,13 @@
-    queries = db.get_queries()
-    return render_template("index.html", queries=queries)
+        searchword = request.args.get('search', '')
+        if len(searchword) > 0:
```
Idiomatic way to do things in python is a simple `if searchword` over using `len` to check for the length of the string

shrayasr commented on app.py:
```diff
@@ -8,8 +8,13 @@
+            return render_template("index.html", queries=queries)
+        else:
+            queries = db.get_queries()
+            return render_template("index.html", queries=queries)
```
Don't duplicate this instruction, it can be common to both branches

shrayasr commented on db.py:
```diff
@@ -39,6 +39,19 @@ def update_query(id, title, sql, tags, desc, who):
+        return who
+    elif len(tags) > 0:
+        return tags
+
```
What happens if all `title`, `who` and `tags` returns no results?

shrayasr commented on db.py:
```diff
@@ -39,6 +39,19 @@ def update_query(id, title, sql, tags, desc, who):
+        return title
+    elif len(who) > 0:
+        return who
+    elif len(tags) > 0:
```
Priority should be `title` > `tags` > `who`

shrayasr commented on templates/index.html:
```diff
@@ -23,7 +23,19 @@
+          <span class="glyphicon glyphicon-search" aria-hidden="true"></span>
+        </button>  
+      </span>
+      </form>
```
Add a clear button too

sarveshms committed:
implementing corrections in lsq search

While doing search if the user enters wrong info. It shows error
info. And clear button enables only after the search parameter is
passed.

sarveshms committed:
implementing code clean up in app, db.py files.

shrayasr commented:
:+1: :shipit:
